src/main/java/gregtech/api/fluids/GTFluidRegistration.java

```diff
@@ -49,7 +52,29 @@ private static void fixFluidRegistryName(@NotNull Fluid fluid, @NotNull String m
                 throw new IllegalStateException("Could not reflect the Forge Master Fluid Registry", e);
             }
         }
-        MASTER_FLUID_REFERENCE.inverse().put(fluid, modid + ':' + fluid.getName());
+        Objects.requireNonNull(MASTER_FLUID_REFERENCE);
+
+        if (DEFAULT_FLUID_NAME == null) {
+            try {
+                Field field = FluidRegistry.class.getDeclaredField("defaultFluidName");
+                field.setAccessible(true);
+                // noinspection unchecked
+                DEFAULT_FLUID_NAME = (BiMap<String, String>) field.get(null);
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                throw new IllegalStateException("Could not reflect the Forge Default Fluid Name map", e);
+            }
+        }
+        Objects.requireNonNull(DEFAULT_FLUID_NAME);
+
+        String masterKey = MASTER_FLUID_REFERENCE.inverse().get(fluid);
+        if (masterKey != null && masterKey.split(":")[0].equals(GTValues.MODID)) {
```

Comment:
I think `masterKey.startsWith(GTValues.MODID)` is probably better here. Same for below with `defaultName`. `String#split` is compiling a regex pattern every time it's called.